Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/DateTimePickerLocalizationTests.cs

```diff
@@ -0,0 +1,393 @@
+            Assert.AreEqual("5.", text);
+        }
+
+        // TimePicker Format String: mm t
+        // TimePicker Time: 1 PM, 59 minute
+        // Separator: ","
+        [Test]
+        public void TimePickerNoHour()
+        {
+            RunningApp.Tap(x => x.Marked("TimePicker"));
+            RunningApp.ClearText("timeFormatString");
+            RunningApp.EnterText("timeFormatString", "mm, t");
+            RunningApp.PressEnter();
+            RunningApp.ClearText("settingTime");
+            RunningApp.EnterText("settingTime", "13, 59");
+            RunningApp.PressEnter();
+            var text = RunningApp.WaitForElement("timeClockOptions")[0].Text;
+            Assert.AreEqual("59, P", text);
+        }
+
+        // TimePicker Format String: first half of uppercase letters
+        // TimePicker Time: 11 PM, 59 minute
+        [Test]
+        public void TimePickerRandomUppercaseBegin()
+        {
+            RunningApp.Tap(x => x.Marked("TimePicker"));
+            RunningApp.ClearText("timeFormatString");
+            RunningApp.EnterText("timeFormatString", "ABCDEGIJLNOP");
+            RunningApp.PressEnter();
+            RunningApp.ClearText("settingTime");
+            RunningApp.EnterText("settingTime", "23, 59");
+            RunningApp.PressEnter();
+            var text = RunningApp.WaitForElement("timeClockOptions")[0].Text;
+            Assert.AreEqual("ABCDEGIJLNOP", text);
+        }
+
+        // TimePicker Format String: latter half of uppercase letters
+        // TimePicker Time: 11 PM, 59 minute
+        [Test]
+        public void TimePickerRandomUppercaseEnd()
+        {
+            RunningApp.Tap(x => x.Marked("TimePicker"));
+            RunningApp.ClearText("timeFormatString");
+            RunningApp.EnterText("timeFormatString", "QRSTUVWXYZ");
+            RunningApp.PressEnter();
+            RunningApp.ClearText("settingTime");
+            RunningApp.EnterText("settingTime", "23, 59");
+            RunningApp.PressEnter();
+            var text = RunningApp.WaitForElement("timeClockOptions")[0].Text;
+            Assert.AreEqual("QRSTUVWXYZ", text);
+        }
+
+        // TimePicker Format String: other lowercase letters
+        // TimePicker Time: 11 PM, 59 minute
+        [Test]
+        public void TimePickerRandomLowercaseLetters()
+        {
+            RunningApp.Tap(x => x.Marked("TimePicker"));
+            RunningApp.ClearText("timeFormatString");
+            RunningApp.EnterText("timeFormatString", "abceijklnopqruvwx");
+            RunningApp.PressEnter();
+            RunningApp.ClearText("settingTime");
+            RunningApp.EnterText("settingTime", "23, 59");
+            RunningApp.PressEnter();
+            var text = RunningApp.WaitForElement("timeClockOptions")[0].Text;
+            Assert.AreEqual("abceijklnopqruvwx", text);
+        }
+
+        // DatePicker Format String: d, M, y
+        // Separator: "/"
+        [Test]
+        public void DatePickerOneDigit()
```

Comment:
It looks like several of these tests follow the same general format - would it be possible to use the [TestCase attribute](https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html) and create fewer test methods?